### PR TITLE
Notify controller of invalid destinations in routes.

### PR DESCRIPTION
**Pull request by plorenz**
Allow xgress and links to be destinguished by ID suffix
If a link destination is missing, send a link fault
Add forget-link router debug agent op to allow testing invalid
destinations

### CODE REVIEW

File: router/forwarder/forwarder.go [hunk 1]
```diff
@@ -129,9 +131,15 @@ func (forwarder *Forwarder) Route(route *ctrl_pb.Route) {
 		circuitFt = newForwardTable()
 	}
 	for _, forward := range route.Forwards {
+		// the initiating Xgress isn't established until we return from routing, so that won't exist yet.
+		// Everything else (links and the terminating xgress) should exist at the time of routing
+		if !strings.HasSuffix(forward.DstAddress, "I") && !forwarder.HasDestination(xgress.Address(forward.DstAddress)) {
```
Comment (dovholuknf) [inline router/forwarder/forwarder.go:136]: magic "I" constant? :/ why not make it a legit constant? just the sort of thing i notice in PRs

File: controller/network/link_controller.go [hunk 2]
```diff
@@ -148,6 +148,7 @@ func (linkController *linkController) missingLinks(routers []*Router, pendingTim
 						if err != nil {
 							return nil, err
 						}
+						id = id + "L"
```
Comment (dovholuknf) [inline controller/network/link_controller.go:151]: "magic" L - :)

File: controller/network/network.go [hunk 3]
```diff
@@ -643,11 +643,13 @@ func (network *Network) CreatePath(srcR, dstR *Router) (*Path, error) {
 	if err != nil {
 		return nil, err
 	}
+	ingressId += "I"
```
Comment (dovholuknf) [inline controller/network/network.go:646]: yah. i'll just say this onelast time..... :)

File: router/forwarder/forwarder.go (hunk 1 above)
Comment (plorenz) [reply router/forwarder/forwarder.go:136]: Makes sense. I'll have to make a  common consts package for it

File: handler_common/common.go [hunk 4]
```diff
@@ -20,7 +22,20 @@ func SendResult(request *channel.Message, ch channel.Channel, message string, su
 
 	response := channel.NewResult(success, message)
 	response.ReplyTo(request)
-	if err := ch.Send(response); err != nil {
+	if err := response.WithTimeout(5 * time.Second).SendAndWaitForWire(ch); err != nil {
```
Comment (dovholuknf) [inline handler_common/common.go:25]: looks like the sort of thing that needs to be doc'ed imo. (the 5s wait)

File: handler_common/common.go [hunk 5]
```diff
@@ -20,7 +22,20 @@ func SendResult(request *channel.Message, ch channel.Channel, message string, su
...
+		log.WithError(err).Error("failed to send result")
+	}
+}
+
+func SendOpResult(request *channel.Message, ch channel.Channel, op string, message string, success bool) {
+	log := pfxlog.ContextLogger(ch.Label()).WithField("operation", op)
+	if !success {
+		log.Errorf("%v error performing %v: (%s)", ch.LogicalName(), op, message)
```
Comment (dovholuknf) [inline handler_common/common.go:33]: should this return here? maybe a bug?

Comment (dovholuknf) [reply handler_common/common.go:33]: oooh nevermind - i see - disregard above.

File: handler_common/common.go [hunk 6]
```diff
@@ -20,7 +22,20 @@ func SendResult(request *channel.Message, ch channel.Channel, message string, su
...
+	}
+
+	response := channel.NewResult(success, message)
+	response.ReplyTo(request)
+	if err := response.WithTimeout(5 * time.Second).SendAndWaitForWire(ch); err != nil {
```
Comment (dovholuknf) [inline handler_common/common.go:38]: are all send and wait's 5s? why not a const?

File: router/handler_ctrl/route.go [hunk 7]
```diff
@@ -83,22 +85,36 @@ func (rh *routeHandler) HandleReceive(msg *channel.Message, ch channel.Channel)
...
 }
 
-func (rh *routeHandler) success(msg *channel.Message, attempt int, route *ctrl_pb.Route, peerData xt.PeerData, log *logrus.Entry) {
-	rh.forwarder.Route(route)
+func (rh *routeHandler) completeRoute(msg *channel.Message, attempt int, route *ctrl_pb.Route, peerData xt.PeerData, log *logrus.Entry) {
+	if err := rh.forwarder.Route(route); err != nil {
+		var ide forwarder.InvalidDestinationError
+		if errors.As(err, &ide) {
+			linkId := string(ide)
+			if strings.HasSuffix(linkId, "L") {
```
Comment (dovholuknf) [inline router/handler_ctrl/route.go:107]: strings.HasSuffix(linkId, "L") seems like it should be a function imo /shrug

**Review by dovholuknf — APPROVED** [review]

File: router/forwarder/forwarder.go (hunk 1 above)
Comment (plorenz) [reply router/forwarder/forwarder.go]: Removed constants and used a type field in the route message instead. Felt cleaner

File: controller/network/network.go (hunk 3 above)
Comment (plorenz) [reply controller/network/network.go]: Fixed. See above.

File: router/handler_ctrl/route.go (hunk 7 above)
Comment (plorenz) [reply router/handler_ctrl/route.go]: Fixed, see above.

File: controller/network/link_controller.go (hunk 2 above)
Comment (plorenz) [reply controller/network/link_controller.go]: Fixed, see above.

**Pull request merged**